rdmd.d

```diff
@@ -813,6 +816,52 @@ addition to compiler options, rdmd recognizes the following options:
 ".format(defaultCompiler, defaultExclusions);
 }
 
+private string discoverCompiler(string requested)
+{
+    // The requested compiler always wins in configuration, regardless of if it can be executed.
+    if (requested.length > 0)
+        return requested;
+    
+    // If we don't have a known compiler specified by the user,
+    // then we need to look to see if it was specified via an environmental argument.
+    // This is used quite heavily within D's ecosystem set the D compiler.
+    // There is some holdouts that still use $DC to set the compiler, so its used as a fallback.
+    // We need to do this due to rdmd being used as the execution engine for shebang files.
+    string environmentSetCompiler = environment.get("DMD", environment.get("DC", null));
+
+    if (environmentSetCompiler.length > 0)
+    {
+        // Absolute files if provided (or relative to CWD), will pass this,
+        // for CI's this will most likely be the outcome.
+        if (Filesystem.existsAsFile(environmentSetCompiler))
+            return environmentSetCompiler;
+        else
+        {
+            // If $DMD/$DC environment arguments are not absolute (or in relative path),
+            // we check if its in $PATH variable by trying to execute it.
+            // This is most likely a much rarer outcome of this function but that is ok.
+            try
+            {
+                auto pid = execute([environmentSetCompiler, "--version"]);
```

Comment:
No, running a process to see if it works is too expensive and dangerous to do as part of auto-detection logic.

Reply:
Okay, all of the checks for environmentSetCompiler can go.

Hopefully, I'm being over-paranoid on that.